### PR TITLE
Streaming: don't decode stream bytes until json.decode

### PR DESCRIPTION
**Pull request by 7diracseas**
Some people (including me) seem to be having issues with UTF-8 encoded tweets when reading a stream. This *should* fix it.

### CODE REVIEW

File: tweepy/streaming.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def read_len(self, length):
             if len(self._buffer) >= length:
                 return self._pop(length)
             read_len = max(self._chunk_size, length - len(self._buffer))
-            self._buffer += self._stream.read(read_len).decode("ascii")
```
Comment (7diracseas) [inline tweepy/streaming.py]: Not really sure why we have to decode as ascii here.

File: tweepy/streaming.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ class ReadBuffer(object):
 
     def __init__(self, stream, chunk_size):
         self._stream = stream
-        self._buffer = u""
+        self._buffer = ''
```
Comment (7diracseas) [inline tweepy/streaming.py]: Keep buffer as bytes until we're ready to decode from JSON.

**Comment by 7diracseas** [comment]
A lot of the tests for the streaming functionality are out of date. I can spend some time fixing those tests as well.

**Comment by joshthecoder** [comment]
Looks good to me. :+1:  Thanks!

**Pull request merged**